src/Bonsai.ML.Visualizers/HeatMapSeriesOxyPlotBase.cs

```diff
@@ -0,0 +1,244 @@
+    {
+        private PlotView view;
+        private PlotModel model;
+        private HeatMapSeries heatMapSeries;
+        private LinearColorAxis colorAxis;
+
+        private ToolStripComboBox paletteComboBox;
+        private ToolStripLabel paletteLabel;
+        private int _paletteSelectedIndex;
+        private OxyPalette palette;
+
+        private ToolStripComboBox renderMethodComboBox;
+        private ToolStripLabel renderMethodLabel;
+        private int _renderMethodSelectedIndex;
+        private HeatMapRenderMethod renderMethod = HeatMapRenderMethod.Bitmap;
+
+        private StatusStrip statusStrip;
+
+        private int _numColors = 100;
+
+        /// <summary>
+        /// Event handler which can be used to hook into events generated when the combobox values have changed.
+        /// </summary>
+        public event EventHandler PaletteComboBoxValueChanged;
+
+        public ToolStripComboBox PaletteComboBox
+        {
+            get => paletteComboBox;
+        }
+
+        public event EventHandler RenderMethodComboBoxValueChanged;
+
+        public ToolStripComboBox RenderMethodComboBox
+        {
+            get => renderMethodComboBox;
+        }
+
+        public StatusStrip StatusStrip
+        {
+            get => statusStrip;
+        }
+
+        /// <summary>
+        /// Constructor of the TimeSeriesOxyPlotBase class.
+        /// Requires a line series name and an area series name.
+        /// Data source is optional, since pasing it to the constructor will populate the combobox and leave it empty otherwise.
+        /// The selected index is only needed when the data source is provided.
+        /// </summary>
+        public HeatMapSeriesOxyPlotBase(int paletteSelectedIndex, int renderMethodSelectedIndex, int numColors = 100)
+        {
+            _paletteSelectedIndex = paletteSelectedIndex;
+            _renderMethodSelectedIndex = renderMethodSelectedIndex;
+            _numColors = numColors;
+            // palette = OxyPalettes.Rainbow(_numColors);
+            Initialize();
+        }
+
+        private void Initialize()
+        {
+            view = new PlotView
+            {
+                Size = Size,
+                Dock = DockStyle.Fill,
+            };
+
+            model = new PlotModel();
+
+            heatMapSeries = new HeatMapSeries {
+                X0 = 0,
+                X1 = 100,
+                Y0 = 0,
+                Y1 = 100,
+                Interpolate = true,
+                RenderMethod = renderMethod,
+                CoordinateDefinition = HeatMapCoordinateDefinition.Edge
+            };
+
+            colorAxis = new LinearColorAxis {
+                Position = AxisPosition.Right,
+            };
+
+            model.Axes.Add(colorAxis);
+            model.Series.Add(heatMapSeries);
+
+            view.Model = model;
+            Controls.Add(view);
+
+            InitilizeColorPalette();
+            InitilizeRenderMethod();
```

Comment:
```suggestion
            InitializeColorPalette();
            InitializeRenderMethod();
```

This class is internal so not critical, but seems like the names have typos here.